include/h2o/memory.h

```diff
@@ -167,8 +167,10 @@ extern void *(*volatile h2o_mem__set_secure)(void *, int, size_t);
 /**
  * prints an error message and aborts
  */
-#define h2o_fatal(msg) h2o__fatal(__FILE__ ":" H2O_TO_STR(__LINE__) ":" msg)
```

Comment:
This is an existing issue, but I'd appreciate it if you could change the code to pass `__FILE__` and `__LINE__` as arguments rather than stringifying them here. The reason is that `__FILE__` is not guaranteed to be a string literal (it may be a `const char *`) and therefore that it should not be concatenated this way. And that we want to process `__FILE__` and `__LINE__` the same way.

Reply:
Yes, Fixed.
BTW,  i'm curious in which case(or environment) `__FILE__` is not string literal but a `const char *`.

Reply:
My bad. I was confused with `__func__`. `__FILE__` is guaranteed to be a string literal.

But no worries. The change makes sense nevertheless. It's better to have a `fatal` function that takes filename and line number as an argument, because it opens up the possibility of reporting a fatal error with the callers call location.